colly.go

```diff
@@ -84,7 +85,7 @@ type HTMLElement struct {
 
 // Context provides a tiny layer for passing data between callbacks
 type Context struct {
-	contextMap map[string]string
+	ContextMap map[string]string
```

Comment:
Why do you make `ContextMap` publicly available? `Context` has a getter/setter which is thread safe.

Reply:
I needed to make `ContextMap` public to make request object serializable with gob.

Reply:
Cannot be `Reqest.Ctx` excluded somehow?